### ISSUE TITLE
Model blow up in case with asymetric mask

### ISSUE THREAD

**Issue by fluidnumericsJoe** [louity/MQGeometry#4]
I'm working on an example that modifies the provided octagonal domain example. Here, I simply change the boundary mask to cut out the lower left corner of the domain. With all other parameters maintained, this introduces an asymmetry to the problem

```
# cut out lower left corner
mask = torch.ones(nx, ny)
for i in range(nx//4):
    for j in range(ny//4):
            mask[i,j] = 0.
```

The solution develops positive vorticity completely around the domain and eventually blows up around year 23.

![Image](https://github.com/user-attachments/assets/3bd9fbcb-82bc-4a80-ba17-dec3fb643344)

I'm open to any suggestions about what could be potentially going wrong here.

**Comment by fluidnumericsJoe** [comment]
In discussion with some colleagues, the asymmetry introduces a positive net vorticity in the domain that is mopped up by an apparent boundary layer. Since the upper layer has no explicit momentum diffusion or drag, it appears MQGeometry can't handle this kind of scenario.
